### PR TITLE
[#49220] Adapt Project-based Boards to be rendered by Rails  

### PR DESCRIPTION
**Pull request by aaron-contreras**
This PR converts the Project Boards Index Page from being rendered by Angular to being rendered by Rails and leveraging our existing `TableComponent`.

- The Angular `NewBoardModalComponent` has been removed and replaced by the Boards form created for use in the global context, making everything but the `boards#show` action be rendered by Rails.
- The `BoardsIndexPageComponent` has been removed as it's no longer in use.

**NOTES:**

- Deleting a board from within a board page's toolbar was a request performed to the API and subsequently a transition to the project's board index page by Angular. In order to have Rails handle the deletion, flash message setting and redirecting to `boards#index` (without needing to re-write this toolbar to be a form or Rails-rendered) added a JSON response that included the `redirect_url` that the component should pick up and set the page's `href`, causing the "Rails-handled redirect". If someone has a better idea on how to accomplish this before the feature freeze or if this is not a correct approach, please do tell.

- While examining the specs for Project based boards, I noticed that there was a bug with the `VersionBoardCreateService`. I've addressed it within this PR as it would pertain to project based boards as well in https://github.com/opf/openproject/commit/fb06f719e0dc88634d86e4933e58e5c45933b2d1 along with an explanation of the bug

See: https://community.openproject.org/work_packages/49220

### CODE REVIEW

File: frontend/src/app/features/boards/board/toolbar-menu/boards-toolbar-menu.directive.ts [hunk 1]
```diff
@@ -103,13 +103,20 @@ export class BoardsToolbarMenuDirective extends OpContextMenuTrigger {
...
-              .then(() => {
-                this.State.go('boards.list', { flash_message: { type: 'success', message: this.text.deleteSuccessful } });
-              });
+            void this.http
+              .delete(
+                `/boards/${this.board.id}`,
+                { responseType: 'json' },
+              )
+              .toPromise()
+              .then(
```
Comment (oliverguenther) [inline frontend/src/app/features/boards/board/toolbar-menu/boards-toolbar-menu.directive.ts]:
```suggestion
              .subscribe(
```

File: frontend/src/app/features/boards/boards-sidebar/boards-menu.component.ts [hunk 2]
```diff
@@ -87,8 +86,12 @@ export class BoardsMenuComponent extends UntilDestroyedMixin implements OnInit {
       });
   }
 
-  showNewBoardModal():void {
-    this.opModalService.show(NewBoardModalComponent, this.injector);
+  redirectToNewBoardForm():void {
+    const newBoardFormPath = (this.currentProject && this.currentProject.identifier)
+      ? `${window.appBasePath}/projects/${this.currentProject.identifier}/boards/new`
```
Comment (oliverguenther) [inline frontend/src/app/features/boards/boards-sidebar/boards-menu.component.ts]: I'd like to have this added to the `PathHelperService`

File: modules/boards/app/controllers/boards/boards_controller.rb [hunk 3]
```diff
@@ -42,15 +47,35 @@ def create
...
     end
 
+    def destroy
+      @board_grid.destroy
+
+      flash[:notice] = I18n.t(:notice_successful_delete)
+
+      respond_to do |format|
+        format.json do
+          render json: { redirect_url: project_work_package_boards_path(@project) }
```
Comment (oliverguenther) [inline modules/boards/app/controllers/boards/boards_controller.rb:64]: This is fine for me 👍

File: modules/boards/config/routes.rb [hunk 4]
```diff
@@ -3,14 +3,18 @@
...
-    get '/boards(/*state)', to: 'boards/boards#index', as: :work_package_boards
+    resources :boards,
+              controller: 'boards/boards',
+              only: %i[index new show],
+              as: :work_package_boards
+
+    # Adding the `create` action into the above `resources` macro would
+    # result in a name collision between it and `get /boards(/*state)`
+    # as it would result in both being named `project_work_package_boards`
+    post '/boards', to: 'boards/boards#create'
```
Comment (oliverguenther) [inline modules/boards/config/routes.rb]: I'm not sure I follow, isn't the naming expected? You did remove the `boards/(*state)` route after all, so `show` now also has the same route helper?

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Behavior looking good to me, there are a few minor remarks.

With this PR, we can remove parts of the angular based action services (for creating new boards at least). I can take a look into that separately.

**Review by oliverguenther — APPROVED** [review]

File: modules/boards/config/routes.rb (hunk 4 above)
Comment (aaron-contreras) [reply modules/boards/config/routes.rb]: It's a stale comment. The issue would arise before I split `boards/(*state)` out into individual `index` and `show` actions. This "special case route" can even be removed an added in into the `resources` macro :)

**Pull request merged**